### ISSUE TITLE
File perm config is ignored

### ISSUE THREAD

**Issue by simonszu** [hlandau/acme#226]
I have a problem with the file permission config as described [here](https://github.com/hlandau/acme/blob/master/_doc/perm.example). 
I want to make the private keys of the certs group-readable for software which is in the acme group, so i created the /var/lib/acme/conf/perm file with the content `keys 0640 0750`. This config file is of course readable by the user acme (i use non-root setup). 
However, after re-running acmetool, the keys folder has still chmod 0700, so i changed the permissions manually to 0750, but after another run of acmetool the permission changed back to 0700. 

So where's my mistake? 
And on a side note, because i got curious: How could my setup ever work if the private keys are all just readable by acme? At least nginx has to read them as well to properly import the cert file into it's vhost config, doesn't it? Where's the black magic behind the normal workflow of acmetool?

**Comment by hlandau** [comment]
I'm unable to replicate this issue. Can you post the output of:
```
$ which acmetool
$ acmetool --version
```

Most daemons read private keys while still root, so the ownership of private keys doesn't matter in that regard.

**Comment by simonszu** [comment]
> $ which acmetool
/usr/bin/acmetool

$ acmetool --version
go version go1.7.3 linux/amd64 gc cgo=true
acmetool Debian version 0.0.58-1

Your answer was a hint for me that i could have used an outdated version of acmetool (although i installed it via apt-get and have apt autoupgrades enabled), so i apt-get upgraded explicitly, and it upgraded from 0.0.56 to 0.0.58. 
I changed the directory permissions of `/var/lib/acme/keys` again to 0750, and started acmetool without any additional commands, so it should just reconcile the keys. 
To my surprise, i have an output now:

> 20161126150851 [WARN] fdb: "keys" has wrong mode -rwxr-x---, changing to -rwx------
20161126150851 [WARN] fdb: "keys/3jg2exxqm6hpjcuqur3tcgnt7ojiupicdybeh4yfcgo53kaqkbza" has wrong mode -rwxr-x---, changing to -rwx------
20161126150851 [WARN] fdb: "keys/3jg2exxqm6hpjcuqur3tcgnt7ojiupicdybeh4yfcgo53kaqkbza/privkey" has wrong mode -rwxr-x---, changing to -rw-------

Repeat the last two lines for all other keys i have in my privkey folder. 

But the additional output made me think about file permission support. Maybe it was only introduced in acmetool versions after 0.0.56? But why is it still ignored? At this point, i have to say i created the file permission config file with copypasting all the commented out text from the documentation file (to have the documentation ready at my hands when i need it), and just adding additional lines beneath it (not commented out) with the actual desired permissions. 
So, out of curiosity, i removed all the commented out text, so the line count of this file is now actually just 1, with the permission i cited at the initial post of this issue. 
Now, it seems that i have resolved this issue, since now, the permissions were applied. But why was this config ignored when i copypasted the additional documentation lines above it? 
Maybe it's an issue with the file parser, maybe it is not reproducible, but at least, this issue is resolved for me.

**Comment by simonszu** [comment]
Ok, i have to reopen this. I have made some new experiments why changing permissions fail (since commented out lines should have no impact).

I am deploying acmetool plus config to the actual server with ansible, and i have made the discovery that acmetool ignored the perm config file after a fresh run of the ansible playbook (which obviously deploys the perm config file to the server as well). 
But after i open this file manually on the server with vim, and saving it to disk again (with no changes), the file isn't ignored anymore and the permissions are applied. At first i suspected some kind of file permission issue, but no. I have no idea what could cause this behaviour. 

I have copied the file permissions after the ansible run and unsuccessful run of acmetool, and the permissions after saving the file as root again and a successful run of acmetool in [this paste](http://pastebin.com/NXis0CkE), but as i said - no idea what could cause this.
